config/templates/stack-template.json

```diff
@@ -438,6 +455,23 @@
         ],
         "UserData": "{{ .UserDataController }}"
       },
+  {{ if .Experimental.AwsEnvironment.Enabled }}
+      "Metadata" : {
+        "AWS::CloudFormation::Init" : {
+          "config" : {
+            "commands": {
+              "write-environment": {
+                "command": { "Fn::Join" : ["", [ "echo '",
+{{range $variable, $function := .Experimental.AwsEnvironment.Environment}}
+"{{$variable}}=", {{$function}} , "\n",
+{{end}}
+"' > /etc/aws-environment" ] ] }
+              }
+            }
+          }
+        }
+      },
+  {{end}}
```

Comment:
Maybe :-) I was actually thinking about these variables in the context of the node pools (so each pool of nodes gets their own set of variables). Also, i'd like to add a etcd autoscaling node group based on @crewjam's work which would need these template variables. TBD